homeassistant/components/azure_event_hub/__init__.py

```diff
@@ -1,89 +1,213 @@
+_LOGGER = logging.getLogger(__name__)
 
-CONF_EVENT_HUB_NAMESPACE = "event_hub_namespace"
-CONF_EVENT_HUB_INSTANCE_NAME = "event_hub_instance_name"
-CONF_EVENT_HUB_SAS_POLICY = "event_hub_sas_policy"
-CONF_EVENT_HUB_SAS_KEY = "event_hub_sas_key"
-CONF_FILTER = "filter"
+QUEUE_BACKLOG_SECONDS = 30
+PAUSE_WHEN_EMPTY = 5
+BATCH_TIMEOUT = 1
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
             {
-                vol.Required(CONF_EVENT_HUB_NAMESPACE): cv.string,
-                vol.Required(CONF_EVENT_HUB_INSTANCE_NAME): cv.string,
-                vol.Required(CONF_EVENT_HUB_SAS_POLICY): cv.string,
-                vol.Required(CONF_EVENT_HUB_SAS_KEY): cv.string,
-                vol.Required(CONF_FILTER): FILTER_SCHEMA,
-            }
+                vol.Optional(CONF_EVENT_HUB_CON_STRING): cv.string,
+                vol.Optional(CONF_EVENT_HUB_NAMESPACE): cv.string,
+                vol.Optional(CONF_EVENT_HUB_INSTANCE_NAME): cv.string,
+                vol.Optional(CONF_EVENT_HUB_SAS_POLICY): cv.string,
+                vol.Optional(CONF_EVENT_HUB_SAS_KEY): cv.string,
+                vol.Optional(CONF_FILTER, default={}): FILTER_SCHEMA,
+            },
+            cv.has_at_least_one_key(
+                CONF_EVENT_HUB_CON_STRING, CONF_EVENT_HUB_NAMESPACE
+            ),
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-async def async_setup(hass: HomeAssistant, yaml_config: Dict[str, Any]):
+async def async_setup(hass, yaml_config):
     """Activate Azure EH component."""
     config = yaml_config[DOMAIN]
-
-    event_hub_address = (
-        f"amqps://{config[CONF_EVENT_HUB_NAMESPACE]}"
-        f".servicebus.windows.net/{config[CONF_EVENT_HUB_INSTANCE_NAME]}"
-    )
-    entities_filter = config[CONF_FILTER]
-
-    client = EventHubClientAsync(
-        event_hub_address,
-        debug=True,
-        username=config[CONF_EVENT_HUB_SAS_POLICY],
-        password=config[CONF_EVENT_HUB_SAS_KEY],
+    entities_filter = config.get(CONF_FILTER)
+    if config.get(CONF_EVENT_HUB_CON_STRING, None):
+        client_args = {"conn_str": config[CONF_EVENT_HUB_CON_STRING]}
+        conn_str_client = True
+    else:
+        client_args = {
+            "fully_qualified_namespace": f"{config[CONF_EVENT_HUB_NAMESPACE]}.servicebus.windows.net",
+            "credential": EventHubSharedKeyCredential(
+                policy=config[CONF_EVENT_HUB_SAS_POLICY],
+                key=config[CONF_EVENT_HUB_SAS_KEY],
+            ),
+            "eventhub_name": config[CONF_EVENT_HUB_INSTANCE_NAME],
+        }
+        conn_str_client = False
+
+    instance = hass.data[DOMAIN] = AEHThread(
+        hass, client_args, conn_str_client, entities_filter
     )
-    async_sender = client.add_async_sender()
-    await client.run_async()
-
-    encoder = JSONEncoder()
-
-    async def async_send_to_event_hub(event: Event):
+    instance.async_initialize()
+
+    async def async_unload_entry(hass: HomeAssistant):
```

Comment:
Please name this something else to not confuse it with our config entry api function.

The argument to this callback will not be `hass` but the event.